src/main/java/org/jenkinsci/plugins/ewm/facets/WorkspaceBrowserFacet.java

```diff
@@ -0,0 +1,54 @@
+package org.jenkinsci.plugins.ewm.facets;
+
+import hudson.model.Fingerprint;
+import jenkins.model.FingerprintFacet;
+import org.jenkinsci.plugins.ewm.model.ExternalWorkspace;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * TODO JAVADOC
+ *
+ * @author Alexandru Somai
+ */
+public class WorkspaceBrowserFacet extends FingerprintFacet {
+
+    public WorkspaceBrowserFacet(@Nonnull Fingerprint fingerprint, long timestamp) {
+        super(fingerprint, timestamp);
+    }
+
+    /**
+     * TODO JAVADOC
+     *
+     * @return
+     */
+    @Restricted(NoExternalUse.class)
+    @SuppressWarnings("unused")
+    public String getWorkspaceLink() {
+        ExternalWorkspace workspace = getWorkspace();
+        if (workspace == null) {
+            return "#";
+        }
+
+        Fingerprint.BuildPtr original = getFingerprint().getOriginal();
+        if (original == null) {
+            return "#";
+        }
+
+        return original.getJob().getAbsoluteUrl() + original.getRun().getSearchUrl() +
```

Comment:
Yes, you are right, the original run may be deleted.

The solution that you are proposing means that I should have a `RootAction` that holds all the allocated workspaces? There is only one concern that I have about it. I will keep adding `ExternalWorkspace` objects to a list (or map) in this `RootAction`, but currently I don't know a mechanism of removing items from that collection (when the workspace is deleted). So, in time, the `RootAction` may end up having to many items in the collection.
Or is there any other technical solution?

At this point, I am proposing to drop the workspace browsing feature from the fingerprint facet, until I'll find a better technical solution.

The user will be able to browse the workspace by clicking on the run item -> from the drop-down list it selects the _External Workspaces_.